beacon-chain/blockchain/chain_info.go

```diff
@@ -314,7 +313,7 @@ func (s *Service) IsOptimistic(ctx context.Context) (bool, error) {
 	if err == nil {
 		return optimistic, nil
 	}
-	if err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
+	if err != doublylinkedtree.ErrNilNode {
```

Comment:
errors.Is is the preferable approach to checking errors in this manner.

https://pkg.go.dev/errors#pkg-overview




```suggestion
	if !errors.Is(err, doublylinkedtree.ErrNilNode) {
```